src/astar/rocstar.js

```diff
@@ -218,22 +222,24 @@ const main = async () => {
 
     console.log(`\nAfter execution, Proxy’s balance is ${chalkPipe('green')(bnToFloat(endProxyBalance.free, decimalBN))} ${symbol}. The delta of proxy balance, or the XCM fee cost is ${chalkPipe('green')(bnToFloat(proxyBalanceDelta, decimalBN))} ${symbol}.`);
 
-    console.log('\n5. Cancel the task ...');
-    const cancelTaskExtrinsic = turingHelper.api.tx.automationTime.cancelTask(taskId);
-    await sendExtrinsic(turingHelper.api, cancelTaskExtrinsic, keyPair);
+    if (schedule.Recurring || schedule.Fixed?.executionTimes.length > 1) {
```

Comment:
The idea is good, but you need to print logs instead of skipping this silently. Add an else and console.log like below.

```
else {
console.log("There is no more task to trigger; skipping task cancellation ...");
} 
```